### DISCUSSION TITLE
Homerow only option

### DISCUSSION THREAD

**Discussion by bobby-sills** [monkeytypegame/monkeytype#3605] [Ideas] [answered]
I think it would be good if there was an option to only type a certain set of keys. This will be good for learning new layouts as you can practice on certain at a time. You could have an option for only the home row and a customizable option where you can choose the letters you want. It would then find the words with only those letters in them. This sounds like it would be complicated to implement though. I think it would be cool though.

**Comment by acxz** [comment]
A quick search shows this result in the codebase: https://github.com/monkeytypegame/monkeytype/blob/092d513f01ad91cfa36eb5c20a67291bd895aebe/frontend/src/ts/modals/word-filter.ts#L64

If you choose `custom` in the UI bar above the words to type, you can change the word filter to just be homerow words.

**Reply by acxz** [reply]
Here is the list of words that contain only homerow keys (qwerty), which you can copy paste into text box for `custom`. Hint make sure to change the mode to `random` and to change the limit.

```
aa ad ag ah al as fa ha ka la sh aah aal aas add ads aff aga aha ala all als ash ask ass dad dag dah dak dal fad fag fas gad gag gal gas had hag hah haj has jag kaf kas lad lag las sad sag sal sha shh ska aahs aals adds agas agha alas alfa alga alls asks dada dads daff dags dahl dahs daks dals dash dhak dhal fads fags fall fash flag flak gads gaff gaga gags gala gall gals gash glad haaf hadj hags haha hahs hajj half hall hash jagg jags kaas kafs kaka khaf lads lags lakh lall lash lass saga sags sall sals sash sass shad shag shah skag skas slag aghas alfas algal algas dadas daffs dagga dahls dhaks dhals falls flags flash flask gaffs galah galas galls glads glass haafs hadal hahas halls jaggs kakas kasha khafs lakhs lalls sagas salad salal salsa shads shags shahs shall skags skald slags slash alaska daggas fallal flasks galahs halala hallah kashas salads salals salsas skalds alaskas alfalfa ashfall fallals haggada halakah halakha halalah halalas hallahs alfalfas ashfalls haggadah haggadas halakahs halakhas halalahs haggadahs
```

Source: https://www.maltron.com/word-lists---qwerty-layout.html

**Comment by Me24Jeff** [comment]
I like the idea and I think that what @acxz suggested is a good way of doing this, but I think it would be nice to have it as a funbox option so I can easily use it for different layouts/languages instead of looking up all the words and then putting it in custom

**Comment by Miodec** [comment, marked as answer]
You can use custom mode > words filter > homerow preset.
